src/application/extension-popup/components/top-bar.tsx

```diff
@@ -0,0 +1,51 @@
+import { useLocation, useNavigate } from 'react-router';
+
+import { IDRISS_ICON_WITH_TEXT } from 'shared/idriss';
+import { IconButton, Toggle } from 'shared/ui';
+import { EXTENSION_POPUP_ROUTE, useExtensionSettings } from 'shared/extension';
+
+export const TopBar = () => {
+  const { extensionSettings, changeExtensionSetting } = useExtensionSettings();
+  const navigate = useNavigate();
+  const { pathname } = useLocation();
+  const isHomeView = pathname === EXTENSION_POPUP_ROUTE.HOME;
```

Comment:
I am a big fan of hiding implementation details, this would be a cool improvement to not couple whole repository with `react-router` interface.

To achieve it you would need to add it inside `usePopup` hook:
```ts
popup.navigate()
popup.currentRoute
```

and not use useNavigate and useLocation anywhere else in code, by doing so we will apply dependency inversion so we have a single interface owned by us instead of being ruled by a library.

On the other hand I know that we will probably never move away from this react-router, but this change shoulnd't be a lot of work, let me know WDYT, I am also fine with keeping react-router coupling

Reply:
sounds good, we won't have to add the `useNavigate`/`useLocation` imports everytime

Reply:
I've also moved the MemoryRouter to the PopupContetxtProvider, let me know what do you think about that approach

Reply:
sounds good!